cpp/src/fil/fil.cu

```diff
@@ -381,6 +398,9 @@ struct dense_forest : forest {
                                num_nodes * sizeof(dense_node),
                                cudaMemcpyHostToDevice,
                                h.get_stream()));
+
+    dispatch_on_fil_template_params(enable_smem_carveout<dense_storage>(max_shm_),
+                                    (predict_params)class_ssp_);
```

Comment:
Why is a type cast needed here?

Reply:
because the type is used to resolve a template instantiation, and for template resolutions, implicit conversions are not considered.